pkg/safe/state.go

```diff
@@ -283,10 +284,27 @@ func (l *List[T]) Find(fn func(T) bool) (T, bool) {
 }
 
 // Iterator returns a new iterator over the list.
+//
+// Deprecated: use [List.Iter] instead.
 func (l *List[T]) Iterator() ListIterator[T] {
 	return ListIterator[T]{pos: 0, list: *l}
 }
 
+// Iter returns a new rangefunc iterator over the list.
+func (l *List[T]) Iter() iter.Seq[T] {
+	return func(yield func(T) bool) {
+		iterator := ListIterator[T]{pos: 0, list: *l}
+
+		for iterator.Next() {
+			res := iterator.Value()
+
+			if !yield(res) {
+				return
+			}
+		}
+	}
+}
+
 // ListIterator is a generic iterator over resource.Resource slice.
 type ListIterator[T any] struct {
```

Comment:
Needs deprecation notice.

Reply:
can do, but also feel free to take this over if you want to push this faster forward.

Reply:
Okay, I'll take it.

Reply:
🙏